script.js

```diff
@@ -1,4 +1,176 @@
+    var winner = "Player 1";
+  } else {
+    winner = "Player 2";
+  }
+  return winner;
+  }; 
+
+// Compare reverse-to-lowest function: Determine lower number as winner
+var getLowestWinner = function (){
+  if (p1Number <= p2Number){
+    var winner = "Player 1";
+  } else {
+    winner = "Player 2";
+  }
+  return winner;
+  };  
+
+// Leaderboard function
+var displayLeaderboard = function () {
+  if (p1Score >= p2Score) {
+    return (
+      "Leaderboard: <br> Player 1 Score: " +
+      p1Score +
+      "<br> Player 2 Score: " +
+      p2Score
+    );
+  } else {
+    return (
+      "Leaderboard: <br> Player 2 Score: " +
+      p2Score +
+      "<br> Player 1 Score: " +
+      p1Score
+    );
+  }
+};
+
 var main = function (input) {
-  var myOutputValue = 'hello world';
+  if (gameMode == gameMode_DiceRoll) {
+    if (currPlayer == 1) {
+      p1DiceArray.push(getDiceRoll());
+      console.log("P1 D1: " + p1DiceArray[0]);
+      p1DiceArray.push(getDiceRoll());
+      console.log("P1 D2: " + p1DiceArray[1]);
+      gameMode = gameMode_ChooseDiceOrder;
+      return (
+        `Hi Player ${currPlayer}, your results are: <br> Roll 1: ${
+        p1DiceArray[0]}
+        <br> Roll 2: ${
+        p1DiceArray[1] }
+        <br><br> Please key in '1' if you want Dice Roll 1 to as first digit or '2' if you want Dice Roll 2 as first digit.`
+      );
+    }
+    if (currPlayer == 2) {
+      p2DiceArray.push(getDiceRoll());
+      console.log("P2 D1: " + p2DiceArray[0]);
+      p2DiceArray.push(getDiceRoll());
+      console.log("P2 D2: " + p2DiceArray[1]);
+      gameMode = gameMode_ChooseDiceOrder;
+      return (
+        `Hi Player ${currPlayer}, your results are: <br> Roll 1: ${
+          p2DiceArray[0]}
+          <br> Roll 2: ${
+          p2DiceArray[1] }
+          <br> Please key in '1' if you want Dice Roll 1 to as first digit or '2' if you want Dice Roll 2 as first digit.`
+        );
+    }
+  }
+
+  if (gameMode == gameMode_ChooseDiceOrder) {
+    if (input == "1") {
+      if (currPlayer == 1) {
+        p1Number = concatenate2Numbers(p1DiceArray[0], p1DiceArray[1]);
+      }
+      if (currPlayer == 2) {
+        p2Number = concatenate2Numbers(p2DiceArray[0], p2DiceArray[1]);
+      }
+    } else if (input == "2") {
+      if (currPlayer == 1) {
+        p1Number = concatenate2Numbers(p1DiceArray[1], p1DiceArray[0]);
+      }
+      if (currPlayer == 2) {
+        p2Number = concatenate2Numbers(p2DiceArray[1], p2DiceArray[0]);
+      }
+    }
+
+    if (currPlayer == 2) {
+      gameMode = gameMode_CompareResults;
+      myOutputValue =
+        `Your chosen number is ${p2Number} <br> Click Submit to find out which Player won!`;
+    }
+
+    if (currPlayer == 1) {
+      gameMode = gameMode_DiceRoll;
+      console.log("changing currPlayer to 2");
+      currPlayer = 2;
+
+      myOutputValue =
+        "Your chosen number is " +
+        p1Number +
+        ". Player 2 click Submit to roll your dice";
+      }
+  }
```

Comment:
Likewise, all the logic following the gameMode conditionals in your main function could be put into its own helper functions